[Experimental] Add e2e tests for the rating filter block

samueljseay opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/42253

Adds basic e2e tests for the rating filter block. These are basic smoke tests, but could be extended in future to cover more complex cases. 

Also a minor rename of a variable in the rating filter block class to match it's intent.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

1. See the new e2e tests pass in CI

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

[Experimental] Add e2e tests for the new product rating filter block.

</details>

samueljseay committed:
Fix up variable naming.

samueljseay committed:
Remove redundant assignment.

samueljseay committed:
Add WIP basic test of the rating filter block.

samueljseay committed:
Merge remote-tracking branch 'origin/trunk' into dev/e2e-test-rating-filter

samueljseay committed:
Add additional reviews to a separate product to show more review options in the e2e test.

samueljseay committed:
Add a test of filtering down the products.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

samueljseay committed:
Remove redundant scroll.

dinhtungdu commented on plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/rating-filter.block_theme.spec.ts:
```diff
@@ -0,0 +1,51 @@
+			page,
+		} ) => {
+			await page.goto( '/product-filters-rating-block/' );
+			const ratingStars = page.locator(
+				'.wc-block-components-product-rating__stars'
+			);
```
Can we replace css class locator here (and below) with a role base or label base locator? This particular element has `aria-label` set to `Rated %d out of 5`. By doing so, if we refactor the markup of stars we don't need to update this test.

dinhtungdu reviewed (commented):
The code looks good and the test is also passing. I just left a comment about the locator. Otherwise, this is LGTM.

samueljseay committed:
Add aria labels to basic elements of the checkbox list and rating filter stars.

samueljseay committed:
Rely on labels to find elements in rating e2e tests.

samueljseay commented:
@dinhtungdu I have refactored selectors to use roles and added opt-in aria label support to CheckboxList so it can also be selected. Can you take another look?

dinhtungdu reviewed (dismissed):
@samueljseay thank you for the update 🙇🏽 ! This is looking good to me. :shipit:

dinhtungdu commented:
Looks like the new rating added in this PR affect other tests.

samueljseay commented:
@dinhtungdu thanks for spotting that. Will fix.

samueljseay committed:
Update review data for reviews e2e tests to reflect the data we create in init.

samueljseay committed:
Change which review is checked, use length - 1 to select the last element always.

samueljseay committed:
Improve fixture data, improve assertions where possible.

samueljseay committed:
Merge remote-tracking branch 'origin/trunk' into dev/e2e-test-rating-filter

dinhtungdu reviewed (approved):
> but i think in future the real solution would be having these fixtures defined in a json or js file and actually having that file used to create the fixtures.

I agree. Thanks for your extra thoughts. Can we also consider using REST API to create reviews at test run time (and remove them after)?

With the plan and the test is passing, this PR looks good to me!

samueljseay commented:
> I agree. Thanks for your extra thoughts. Can we also consider using REST API to create reviews at test run time (and remove them after)?

Yes I think now after assessing how easy it is to use rest API we should refactor anything using fixtures to use the rest api to set/tear them down. I think we can do that in future PRs.
